File: web-nightly-r1/currency-mobile-app/android-app/app/src/main/java/com/oxyzenq/currencyconverter/data/model/Currency.kt
package com.oxyzenq.currencyconverter.data.model

import com.google.gson.annotations.SerializedName

data class Currency(
    val code: String,
    val name: String,
    val flag: String
)

data class ConversionRequest(
    @SerializedName("from_currency")
    val fromCurrency: String,
    @SerializedName("to_currency")
    val toCurrency: String,
    val amount: Double
)

data class ConversionResponse(
    val success: Boolean,
    @SerializedName("from_currency")
    val fromCurrency: String,
    @SerializedName("to_currency")
    val toCurrency: String,
    val amount: Double,
    @SerializedName("converted_amount")
    val convertedAmount: Double,
    @SerializedName("exchange_rate")
    val exchangeRate: Double,
    val timestamp: String,
    val error: String?
)

data class ExchangeRatesResponse(
    @SerializedName("base_currency")
    val baseCurrency: String,
    val rates: Map<String, Double>,
    val timestamp: String,
    val source: String
)

data class CurrencyListResponse(
    val currencies: Map<String, String>,
    val count: Int,
    val timestamp: String
)

data class ApiError(
    val error: String,
    val message: String,
    val timestamp: String
)

File: web-nightly-r1/currency-mobile-app/backend/app/models/currency.py
from pydantic import BaseModel, Field
from typing import Dict, Optional
from datetime import datetime

class CurrencyCode(BaseModel):
    code: str = Field(..., min_length=3, max_length=3, description="3-letter currency code")
    name: str = Field(..., description="Full currency name")
    country: str = Field(..., description="Country code")

class ConversionRequest(BaseModel):
    from_currency: str = Field(..., min_length=3, max_length=3, description="Source currency code")
    to_currency: str = Field(..., min_length=3, max_length=3, description="Target currency code")
    amount: float = Field(..., gt=0, le=1000000, description="Amount to convert")

class ConversionResponse(BaseModel):
    from_currency: str
    to_currency: str
    amount: float
    converted_amount: float
    exchange_rate: float
    timestamp: datetime
    formatted_result: str

class ExchangeRatesResponse(BaseModel):
    base_currency: str
    rates: Dict[str, float]
    timestamp: datetime
    source: str = "exchangerate-api"

class CurrencyListResponse(BaseModel):
    currencies: Dict[str, str]  # code -> country_code mapping
    count: int
    timestamp: datetime

class HistoricalRateRequest(BaseModel):
    base_currency: str = Field(..., min_length=3, max_length=3)
    target_currency: str = Field(..., min_length=3, max_length=3)
    date: str = Field(..., description="Date in YYYY-MM-DD format")

class APIError(BaseModel):
    error: str
    message: str
    timestamp: datetime
